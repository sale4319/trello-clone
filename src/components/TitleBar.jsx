import React from 'react';
import { useHistory } from 'react-router';
import { Routes } from '../constants/Routes';

export const TitleBar = () => {
    const history = useHistory();

    return (
        <div className="titlebar-container">
            <img className="titlebar-image" src="/home.png" alt="Home" onClick={() => history.push(Routes.Home)} />&nbsp;
            <img className="titlebar-image" src="/boards.png" alt="Boards" onClick={() => history.push(Routes.Boards)} />
            <div className="titlebar-title">Trello's Fratello</div>
        </div>
    );
};