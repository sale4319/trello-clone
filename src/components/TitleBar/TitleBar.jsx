import React from 'react';
import { useHistory } from 'react-router';
import { Routes } from '../../constants/Routes';

const TitleBar = () => {
    const history = useHistory();

    return (
        <div className="titlebar-container">
            <div className="titlebar-image-container">
                <img data-testid="img" className="titlebar-image" src="/boards.png" alt="Boards" onClick={() => history.push(Routes.Boards)} />
            </div>
            <div data-testid="title" className="titlebar-title">Trello's Fratello</div>
        </div>
    );
};

export default TitleBar;