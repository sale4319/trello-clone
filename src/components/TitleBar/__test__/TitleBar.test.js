import React from 'react';
import { TitleBar }from '../TitleBar';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test("image renders with correct alt and src", () => {
    const component = render(<TitleBar />);
    const imgEl = component.getByTestId("img");

    expect(imgEl.alt).toBe("Boards");
    expect(imgEl.src).toEqual("http://localhost/boards.png");
});

test("title renders with correct text", () => {
    const component = render(<TitleBar />);
    const titleEl = component.getByTestId("title");

    expect(titleEl.textContent).toBe("Trello's Fratello");
});