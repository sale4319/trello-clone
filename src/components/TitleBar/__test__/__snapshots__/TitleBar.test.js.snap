// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mateches snapshot 1`] = `
<div
  className="titlebar-container"
>
  <div
    className="titlebar-image-container"
  >
    <img
      alt="Boards"
      className="titlebar-image"
      data-testid="img"
      onClick={[Function]}
      src="/boards.png"
    />
  </div>
  <div
    className="titlebar-title"
    data-testid="title"
  >
    Trello's Fratello
  </div>
</div>
`;
